### ISSUE TITLE
poweropti Firmware 2.01.32 - Verbindung verweigert

### ISSUE THREAD

**Issue by MILUX-DE** [pschlan/powerfox_prometheus_exporter#1]
"Es konnte keine Verbindung hergestellt werden, da der Zielcomputer die Verbindung verweigerte"

poweropti Firmware: 2.01.32

**Comment by pschlan** [comment]
Ja, die neuen Firmware-Versionen erlauben nur noch Zugriff per HTTPS mit einem Zertifikat / Private Key. Die Änderungen dafür sind recht simpel, und das Zertifikat kann man sich aus der Android-App (.apk) rausfischen.

Ich kan meine Änderungen dazu mal pushen, sobald ich Zeit finde...

**Comment by cablehub** [comment]
@pschlan Ich hätte interesse daran, habe hier einen PowerOpti, den ich gerne abfragen würde. Könntest du die notwendigen Änderungen ins Repo pushen?

**Comment by pschlan** [comment]
Sorry, ich hab das Teil in die Tonne gehauen wegen der für mich inakzeptablen Geschäftspraktiken des Herstellers. Kann da leider nicht mehr helfen.

**Comment by cablehub** [comment]
Hi Patrick,
danke für die Rückmeldung. Kannst du mir zumindest sagen, ob das Auslesen trotz neuerer Firmware bei dir letztlich geklappt hat? Ich habe in der APK ein Zertifikat gefunden, und wenn ich das bei einer `curl`-Abfrage mit übergebe, kommt zwar kein Zertifikatsfehler mehr, aber es kommen auch keine Daten zurück.

Auf was für eine Lösung bist du umgestiegen? Ich suche nach einer Lösung, das Smart-Meter auszulesen, um eine Wallbox nach Einspeiseleistung zu steuern. Am liebsten wäre mir etwas komplett ohne Cloud (habe bspw. noch einen Raspberry Pi herumliegen, der Auslesen und Steuern übernehmen könnte).

**Comment by pschlan** [comment]
Hi, ja, das hatte damals zumindest noch geklappt... Hast du mal curl verbose laufen lassen und geschaut, ob es irgend welche Fehler ausgibt? Ich meine mich dunkel zu erinnern, dass man noch ein Passwort für das Cert brauchte. (War aber hartkodiert in der App.) Vllt. finde ich ja auch noch meinen alten Code wieder. 
Momentan nutze ich Tibber mit deren Auslesegerät. Die haben eine ordentliche API und für den Stromtarif müssen die Daten eh in die Cloud...
